### ISSUE TITLE
Docs for custom data format options `encoding` and `read`

### ISSUE THREAD

**Issue by zachleat** [11ty/eleventy#2379]
Needs to be added to the docs: 

```
Allow custom data extensions to have options!

* `read: false` which skips the file contents from being read by Eleventy. This mades the argument passed to the data extension callback a file path instead of the file contents.
* `encoding` which allows you to override the file encoding passed to fs.readFile (specifically I wanted to use `null` for Buffer)
```

**Comment by nhoizey** [comment]
@zachleat I'm reading https://www.11ty.dev/docs/data-custom/#feed-exif-image-data-into-the-data-cascade now! 😍

I'm wondering: do you cache the result of `exifr` readings between builds?

I'm also using `exifr` for my photography site (for example [this page](https://nicolas-hoizey.photo/galleries/travels/europe/georgia/overlooking-tbilisi/)), but asynchronously, generating new Markdown files (like [this one](https://raw.githubusercontent.com/nhoizey/nicolas-hoizey.photo/main/src/photos/overlooking-tbilisi/index.md)) only each time I add a new photo, because `exifr` takes time (and RAM) with heavy photos (like [this 3.5MB one](https://github.com/nhoizey/nicolas-hoizey.photo/blob/main/src/photos/overlooking-tbilisi/overlooking-tbilisi.jpg)).

**Comment by nhoizey** [comment]
I agree for `--incremental` priority, even if I've never seen a convincing implementation, in any tool… 😅

I'm doing a lot of things with the photos, not just using `exifr`, and I'm parsing all photos every time I run the script, to check if there are some new EXIF/IPTC data. I guess I should first check the file modification date/time and keep track of it.

Here's the Node script: https://github.com/nhoizey/nicolas-hoizey.photo/blob/main/_scripts/sync-photos.js
